core/commands/storage.go

```diff
@@ -216,11 +214,16 @@ Receive proofs as collateral evidence after selected nodes agree to store the fi
 				if err != nil {
 					return err
 				}
+				// use host askingPrice instead if provided
+				if ni.StoragePriceAsk != 0 {
```

Comment:
should validate this when >= 0 instead since we allow a host set their ask price to 0

Reply:
really? i think if price is zero, that's a special case won't need involve payment side, can communication can be simplify a lot. We haven't implement that, I think it's a special case we can handle better, shouldn't use the same looping way.

Reply:
we still need to involve payment even the host compensation is zero, cause it still need to trigger escrow to sign the contract and let guard challenge/repair during the term.